File: 1408-string-matching-in-an-array/1408-string-matching-in-an-array.cpp
class Solution {
public:
    vector<string> stringMatching(vector<string>& words) {
        sort(words.begin(), words.end(), [](const string& a, const string& b) {
            return a.size() < b.size();
        });
        
        vector<string> result;
        for (const auto& word : words) {
            for (const auto& w : words) {
                if (w != word && w.find(word) != string::npos) {
                    result.push_back(word);
                    break;
                }
            }
        }
        
        return result;
    }
};

File: 1408-string-matching-in-an-array/1408-string-matching-in-an-array.rs
impl Solution {
    pub fn string_matching(words: Vec<String>) -> Vec<String> {
        let mut results = vec![];
        for word in &words {
            if words.iter().any(|x| x != word && x.contains(word)) {
                results.push(word.clone());
            }
        }
        results
    }
}
